src/main/java/com/starrocks/connector/flink/table/source/StarRocksExpressionExtractor.java

```diff
@@ -88,7 +88,10 @@ public String visit(CallExpression call) {
 
     @Override
     public String visit(ValueLiteralExpression valueLiteral) {
-        if (valueLiteral.getOutputDataType() == DataTypes.DATE() || valueLiteral.getOutputDataType().toString().equals("DATE NOT NULL")) {
+        if (valueLiteral.getOutputDataType().getLogicalType().getTypeRoot().equals(DataTypes.TIMESTAMP().getLogicalType().getTypeRoot()) || 
+            valueLiteral.getOutputDataType().getLogicalType().getTypeRoot().equals(DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE().getLogicalType().getTypeRoot()) || 
+            valueLiteral.getOutputDataType().getLogicalType().getTypeRoot().equals(DataTypes.TIMESTAMP_WITH_TIME_ZONE().getLogicalType().getTypeRoot()) ||
+            valueLiteral.getOutputDataType().getLogicalType().getTypeRoot().equals(DataTypes.DATE().getLogicalType().getTypeRoot())) {
             return "'" + valueLiteral.toString() + "'";
```

Comment:
What if the valueLiteral's type is among [`VARCHAR`, `CHAR`, ...]

Reply:
If the valueLiteral's type is among [VARCHAR, CHAR, ...]，The value of the expression is automatically enclosed in single quotes